app/repositories/Repository.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+import java.sql.Timestamp
+import java.time.LocalDateTime
+
+import javax.inject.Inject
+import models.paye.PayeAtsMiddleTier
+import org.slf4j.LoggerFactory
+import play.api.libs.json.{JsObject, Json, Reads, __}
+import play.modules.reactivemongo.ReactiveMongoApi
+import reactivemongo.api.indexes.{Index, IndexType}
+import reactivemongo.bson.BSONDocument
+import reactivemongo.play.json.ImplicitBSONHandlers.JsObjectDocumentWriter
+import reactivemongo.play.json.collection.JSONCollection
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import reactivemongo.play.json.collection._
+
+import scala.concurrent.duration.FiniteDuration
+
+class Repository @Inject()(mongo: ReactiveMongoApi) {
+  val logger = LoggerFactory.getLogger("application." + getClass.getCanonicalName)
+
+  private val collectionName: String = "tax-summaries"
+
+  private def collection: Future[JSONCollection] =
+    mongo.database.map(_.collection[JSONCollection](collectionName))
+
+  private val lastUpdatedIndex = Index(
+    Seq("expiresAt" -> IndexType.Ascending),
+    name = Some("expires-at-index"),
+    options = BSONDocument("expireAfterSeconds" -> 0))
+
+  def buildId(nino: String, taxYear: Int): String = s"$nino::$taxYear"
```

Comment:
I am not sure we are allowed to capture nino's as the IDs, you may need to encrypt this cache, it's worth a discussion with security

Reply:
hmmm, we can cache this on sessionId maybe instead  but that may not cover all use cases.  I assumed mongo in protected was secured.  Will Check